packages/react/src/components/Toggletip/__tests__/Toggletip-test.js

```diff
@@ -202,5 +202,245 @@ describe('Toggletip', () => {
+            <ToggletipContent>
+              <button>Action</button>
+            </ToggletipContent>
+          </Toggletip>
+        );
+
+        const actionButton = screen.getByText('Action');
+        actionButton.focus();
+        await userEvent.keyboard('{Escape}');
+
+        expect(screen.getByText('Toggle')).toHaveFocus();
+      });
+    });
+
+    describe('Toggletip Alignment', () => {
+      it.each([
+        'left-start',
+        'left-end',
+        'right-start',
+        'right-end',
+        'top-start',
+        'top-end',
+        'bottom-start',
+        'bottom-end',
+      ])('should handle %s alignment correctly', (alignment) => {
+        const { container } = render(
+          <Toggletip align={alignment} defaultOpen>
+            <ToggletipButton>Toggle</ToggletipButton>
+            <ToggletipContent>Content</ToggletipContent>
+          </Toggletip>
+        );
+
+        expect(container.firstChild).toHaveClass(`cds--popover--${alignment}`);
+      });
+    });
+
+    describe('Toggletip Closing Behavior', () => {
+      it('should not close when clicking inside the toggletip during auto-alignment', async () => {
+        render(
+          <Toggletip autoAlign defaultOpen>
+            <ToggletipButton>Toggle</ToggletipButton>
+            <ToggletipContent>
+              <div data-testid="content">Content</div>
+            </ToggletipContent>
+          </Toggletip>
+        );
+
+        await userEvent.click(screen.getByTestId('content'));
+        expect(screen.getByText('Toggle')).toHaveAttribute(
+          'aria-expanded',
+          'true'
+        );
+      });
+
+      it('should close when focus moves outside the toggletip', () => {
+        const { container } = render(
+          <>
+            <button data-testid="external-button">External</button>
+            <Toggletip defaultOpen>
+              <ToggletipButton>Toggle</ToggletipButton>
+              <ToggletipContent>Content</ToggletipContent>
+            </Toggletip>
+          </>
+        );
+
+        const toggleButton = screen.getByText('Toggle');
+        const externalButton = screen.getByTestId('external-button');
+
+        fireEvent.blur(toggleButton, {
+          relatedTarget: externalButton,
+        });
+
+        expect(container.lastChild).not.toHaveClass(
+          `${prefix}--toggletip--open`
+        );
+        expect(container.lastChild).not.toHaveClass(`${prefix}--popover--open`);
+      });
+
+      it('should not close when open and relatedTarget is null', () => {
+        const { container } = render(
+          <Toggletip defaultOpen>
+            <ToggletipButton>Toggle</ToggletipButton>
+            <ToggletipContent>
+              <button>Action</button>
+            </ToggletipContent>
+          </Toggletip>
+        );
+
+        expect(container.firstChild).toHaveClass(`${prefix}--toggletip--open`);
+        const toggletipWrapper = container.firstChild;
+
+        fireEvent.blur(toggletipWrapper, {
```

Comment:
same here

Reply:
Hey Ari , I replaced fireEvent with userEvent on other places
for this test case I used fireEvent instead of userEvent here to explicitly test blur behavior with a `null` relatedTarget. Thoughts ?

Reply:
ah yup! looks good!